compiler/rustc_query_system/src/error.rs

```diff
@@ -77,4 +77,16 @@ pub struct IncrementCompilation {
 
 #[derive(SessionDiagnostic)]
 #[diag(query_system::query_overflow)]
-pub struct QueryOverflow;
+pub struct QueryOverflow {
+    #[subdiagnostic]
+    pub layout_of_depth: Option<LayoutOfDepth>,
+}
+
+#[derive(SessionSubdiagnostic)]
+#[note(query_system::layout_of_depth)]
+pub struct LayoutOfDepth {
+    #[primary_span]
+    pub span: Span,
```

Comment:
It feels like ideally this span would be part of the main message, not of the note. I'm not sure if `derive(SessionDiagnostic)` knows how to deal with `Option<Span>`, but if it doesn't, you can get away with using `DUMMY_SP` for the "default"/"empty" case.

Reply:
We support `Option<Span>` :)

Reply:
That makes sense!